src/ripple/app/misc/Validations.cpp

```diff
@@ -387,6 +387,39 @@ class ValidationsImp : public Validations
         return ret;
     }
 
+    std::uint32_t getValidationTime (uint256 const& hash, int minValidations)
```

Comment:
The problem with returning a pair is that it makes calling awkward. You either use `auto` to keep the pair around and have `second` everywhere you use it or you need three lines of code to use `std::tie`.

Using `boost::optional` is better at the call site, because you can just use `auto` at the call site, `if` to test, and `*` to access. But if the special value is zero and the type is an integer, it's really hard to see any advantage, except that you either need an extra line of code to declare a reference or you have to put a `*` on every access. Using a special return value of zero works almost precisely the same way. The type essentially already is optional and already tests `false` if not present.